run.c

```diff
@@ -544,6 +549,11 @@ int main(int argc, char *argv[]) {
     int token = 1;   // init with token 1 (=BOS), as done in Llama-2 sentencepiece tokenizer
     int pos = 0;     // position in the sequence
     printf("<s>\n"); // explicit print the initial BOS token for stylistic symmetry reasons
+    int size_qkvO = config.dim*config.dim*config.n_layers;
+    int size_ffn = config.hidden_dim*config.dim*config.n_layers;
+    int size_cls = config.dim*config.vocab_size;
+
+    #pragma omp target data map(to:weights.wq[0:size_qkvO],weights.wk[0:size_qkvO],weights.wv[0:size_qkvO],weights.wo[0:size_qkvO],weights.w1[0:size_ffn],weights.w3[0:size_ffn],weights.w2[0:size_ffn],weights.wcls[0:size_cls])
```

Comment:
Nice! Didn't really think it would be a single directive like this.

Reply:
Where does the `weights.` prefix come from?